api/src/Security/OAuth/GoogleAuthenticator.php

```diff
@@ -67,6 +67,9 @@ public function authenticate(Request $request): Passport {
                     $profile->surname = $googleUser->getLastName();
                     $user = new User();
                     $user->profile = $profile;
+                }
+
+                if (in_array($user->state, [null, User::STATE_NONREGISTERED, User::STATE_REGISTERED])) {
```

Comment:
Looks very clean now. But you have an early return above, so if `$existingProfile` is true, you will never get here.

I think the best is if you extract the logic above into a separate function (e.g. findOrCreateUser) and then you could keep the cleanliness of this last state-check.

Reply:
An existing Profile has to have run through the If below in a previous request, else it cannot be an existing Profile.
As long as we don't reset the state, this should be good enough

Reply:
Ah, get it. Thanks for clarifying.